src/StoreBase.ts

```diff
@@ -60,178 +56,211 @@ export abstract class StoreBase {
     protected trigger(keyOrKeys?: string|number|(string|number)[]) {
-        let keys: string[]|undefined;
-
-        // trigger(0) is valid, ensure that we catch this case
-        if (keyOrKeys || _.isNumber(keyOrKeys)) {
-            keys = _.map(_.isArray(keyOrKeys) ? keyOrKeys : [keyOrKeys], key => _.isNumber(key) ? key.toString() : key);
+        // If we're throttling, save execution time
+        let throttledUntil: number | undefined;
+        let throttleMs: number | undefined;
+        if (this._throttleMs !== undefined) {
+            throttleMs = this._throttleMs;
+        } else {
+            // If the store doens't define any throttling, pick up the default
+            throttleMs = Options.defaultThrottleMs;
+        }
+        if (throttleMs) {
+            if (!this._throttleData) {
+                // Needs to accumulate and trigger later -- start a timer if we don't have one running already
+                // If there are no callbacks, don't bother setting up the timer
+                this._throttleData = {
+                    timerId: Options.setTimeout(this._resolveThrottledCallbacks, this._throttleMs),
+                    callbackTime: Date.now() + throttleMs
+                };
+            }
+            throttledUntil = this._throttleData.callbackTime;
         }
 
-        // Build a list of callbacks to call, trying to accumulate keys into a single callback set to avoid multiple callbacks
-        // to the same target with different keys.
+        const bypassBlock = this._bypassTriggerBlocks;
 
-        if (!keys) {
+        // trigger(0) is valid, ensure that we catch this case
+        if (!keyOrKeys && !_.isNumber(keyOrKeys)) {
             // Inspecific key, so generic callback call
             const allSubs = _.flatten(_.values(this._subscriptions));
             _.forEach(allSubs, callback => {
-                // Clear the key list to null for the callback
-                this._gatheredCallbacks.set(callback, null);
+                const existingMeta = StoreBase._pendingCallbacks.get(callback);
+                const newMeta = { keys: null, throttledUntil, bypassBlock };
+                // Clear the key list to null for the callback but respect previous throttle/bypass values
+                if (existingMeta && throttledUntil && existingMeta.throttledUntil) {
+                    newMeta.throttledUntil = Math.min(throttledUntil, existingMeta.throttledUntil);
+                }
+                if (existingMeta && existingMeta.bypassBlock) {
+                    newMeta.bypassBlock = true;
+                }
+                StoreBase._pendingCallbacks.set(callback, newMeta);
             });
 
             _.forEach(_.flatten(_.values(this._autoSubscriptions)),
                 sub => {
-                    this._gatheredCallbacks.set(sub.callback, null);
+                    const existingMeta = StoreBase._pendingCallbacks.get(sub.callback);
```

Comment:
This logic looks like copy/paste. How about `.concat` all subscription callbacks and auto-subscriptions callbacks, so you only need one `.forEach`?